pkg/azuredisk/controllerserver_test.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azuredisk
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-10-01/compute"
+)
+
+func TestGetCachingMode(t *testing.T) {
+	tests := []struct {
+		options   map[string]string
+		expected1 compute.CachingTypes
+		expected2 error
+	}{
+		{
+			nil,
+			compute.CachingTypes(defaultAzureDataDiskCachingMode),
+			nil,
+		},
+		{
+			map[string]string{},
+			compute.CachingTypes(defaultAzureDataDiskCachingMode),
+			nil,
+		},
+		{
+			map[string]string{"cachingmode": ""},
+			compute.CachingTypes(defaultAzureDataDiskCachingMode),
+			nil,
+		},
+		{
+			map[string]string{"cachingmode": "None"},
+			compute.CachingTypes("None"),
+			nil,
+		},
+		{
+			map[string]string{"cachingmode": "ReadOnly"},
+			compute.CachingTypes("ReadOnly"),
+			nil,
+		},
+		{
+			map[string]string{"cachingmode": "ReadWrite"},
+			compute.CachingTypes("ReadWrite"),
+			nil,
+		},
+		{
+			map[string]string{"cachingmode": "WriteOnly"},
+			compute.CachingTypes(""),
+			fmt.Errorf("azureDisk - %s is not supported cachingmode. Supported values are %s", "WriteOnly", supportedCachingModes.List()),
```

Comment:
I think check whether there is error is enough, otherwise if the original error changes, unit test will always change.

Reply:
You're right, I will modify it.